pyne/openmc_utils.py

```diff
@@ -156,3 +171,274 @@ def xml(self):
+    rel_err_tot = np.zeros_like(res_tot)
+    std_dev = np.reshape(flux.std_dev.flatten(),
+                         newshape=(num_e_groups, num_ves))
+    std_dev = std_dev.transpose()
+    var_tot = np.sum(np.square(std_dev), axis=1)
+    nonzero = res_tot > 0
+    rel_err_tot[nonzero] = np.sqrt(var_tot[nonzero]) / res_tot[nonzero]
+
+    # get volume averged results
+    result = np.divide(result, ve_vol)
+    res_tot = np.divide(res_tot, ve_vol)
+    # In the result of OpenMC, x changes fastest
+    # change the order to z changes fastest
+    result = result_changes_order(result, m.dims[3:6])
+    rel_err = result_changes_order(rel_err, m.dims[3:6])
+    res_tot = result_changes_order(res_tot, m.dims[3:6])
+    rel_err_tot = result_changes_order(rel_err_tot, m.dims[3:6])
+    return result, rel_err, res_tot, rel_err_tot
+
+def result_changes_order(result, dims):
+    """
+    This function changes the order of openmc flux data. The default order of
+    OpenMC flux is x changes the fastest. This function changes it to the order
+    which z changes the fasest (the same order as MCNP meshtal).
+
+    Parameters:
+    -----------
+    result : numpy array
+        This numpy array contains the flux data read from OpenMC state point
+        file. The shape of this numpy array is (num_ves*num_e_groups) or
+        (num_ves, ), with x changes the fastest.
+    dims : list
+        The dimension of x, y, z.
+
+    Returns:
+    --------
+    result_ : numpy array
+        This numpy array contains the flux data read from OpenMC state point
+        file. The shape of this numpy array is (num_ves*num_e_groups) or
+        (num_ves, ), with changes the fastest.
+    """
+
+    x_dim, y_dim, z_dim = dims[0], dims[1], dims[2]
+    num_e_groups = len(result.flatten()) // (x_dim * y_dim * z_dim)
+    result_ = np.reshape(result, newshape=(z_dim, y_dim, x_dim, num_e_groups))
+    # set energy changes fastest, z next, y next and then x
+    result_ = np.moveaxis(result_, [0, 1, 2, 3], [2, 1, 0, 3])
+    result_ = np.reshape(result_, newshape=(x_dim*y_dim*z_dim, num_e_groups))
+    return result_
+
+
+def create_meshtally(filename, tally_id, mesh_id=None, particle=None,
+                     tag_names=None, mesh_has_mats=False):
+    """
+    This function creates a MeshTally instance from OpenMC statepoint file.
+
+    Parameters:
+    -----------
+    filename : str
+        Filename of the OpenMC statepoint file. It ends with ".h5",
+        eg: "statepoint.10.h5".
+    tally_id : int
+        Tally number.
+    mesh_id : int
+        Mesh id of this tally used. Required if multiple meshes exist in the
+        OpenMC state point file.
+    particle : str
+        The particle type, 'neutron' or 'photon'.
+    tag_names : iterable, optional
+        Four strs that specify the tag names for the results, relative
+        errors, total results and relative errors of the total results.
+    mesh_has_mats: bool
```

Comment:
Given that we throw an error if this is set to `True`, should we even have it as an option here?

Reply:
I suggest to kept it here reminding us to implement it. I have changed the error message about it from `ValueError` to `NotImplementedError`.

Reply:
I'm curious though, in what scenario will we have materials coming from a mesh in an OpenMC state point file?